sklearn/preprocessing/_discretization.py

```diff
@@ -271,16 +275,8 @@ def fit(self, X, y=None, sample_weight=None):
 
             elif self.strategy == "quantile":
                 quantiles = np.linspace(0, 100, n_bins[jj] + 1)
-                if sample_weight is None:
-                    bin_edges[jj] = np.asarray(np.percentile(column, quantiles))
-                else:
-                    bin_edges[jj] = np.asarray(
-                        [
-                            _weighted_percentile(column, sample_weight, q)
-                            for q in quantiles
-                        ],
-                        dtype=np.float64,
-                    )
```

Comment:
We should keep the original code here: when subsampling with weights has happened, `sample_weight` is set to `None` at line 255.

But on small datasets (without subsampling), and with weights, we still need to call `_weighted_percentile`. Otherwise, the `check_sample_weight_equivalence` check should fail (because it runs the `strategy="quantile"` case on small datasets.

Reply:
Oh whoops, somehow forgot about that